services/rfq/guard/service/guard.go

```diff
@@ -247,6 +287,75 @@ func (g Guard) runChainIndexer(ctx context.Context, chainID int) (err error) {
 	return nil
 }
 
+//nolint:cyclop
+func (g Guard) runChainIndexerV2(ctx context.Context, chainID int, handler fastBridgeHandler) (err error) {
+	chainListener := handler.listeners[chainID]
+
+	parser, err := fastbridgev2.NewParser(chainListener.Address())
+	if err != nil {
+		return fmt.Errorf("could not parse: %w", err)
+	}
+
+	err = chainListener.Listen(ctx, func(parentCtx context.Context, log types.Log) (err error) {
+		et, parsedEvent, ok := parser.ParseEvent(log)
+		// handle unknown event
+		if !ok {
+			if len(log.Topics) != 0 {
+				logger.Warnf("unknown event %s", log.Topics[0])
+			}
+			return nil
+		}
+
+		ctx, span := g.metrics.Tracer().Start(parentCtx, fmt.Sprintf("handleLog-%s", et), trace.WithAttributes(
+			attribute.String(metrics.TxHash, log.TxHash.String()),
+			attribute.Int(metrics.Origin, chainID),
+			attribute.String(metrics.Contract, log.Address.String()),
+			attribute.String("block_hash", log.BlockHash.String()),
+			attribute.Int64("block_number", int64(log.BlockNumber)),
+		))
+
+		defer func() {
+			metrics.EndSpanWithErr(span, err)
+		}()
+
+		switch event := parsedEvent.(type) {
+		case *fastbridgev2.FastBridgeV2BridgeRequested:
+			v1Event := &fastbridge.FastBridgeBridgeRequested{
+				TransactionId: event.TransactionId,
+				Sender:        event.Sender,
+				Request:       event.Request,
+				DestChainId:   event.DestChainId,
+				OriginToken:   event.OriginToken,
+				DestToken:     event.DestToken,
+				DestAmount:    event.DestAmount,
+				SendChainGas:  event.SendChainGas,
+				Raw:           event.Raw,
+			}
```

Comment:
Hm, actually since the V2 has all the events from V1 we _could consider_ using the V2 parser and event types everywhere in the guard. This should also dedupe some/most of the V1/V2 almost-identical-functions.

Wdyt @trajan0x?

Reply:
I think this makes it easier to deprecate v1 functionality later tbh